python/storage-initializer/scripts/initializer-entrypoint

```diff
@@ -11,4 +14,15 @@ src_uri = sys.argv[1]
 dest_path = sys.argv[2]
 
 logging.info("Initializing, args: src_uri [%s] dest_path[ [%s]" % (src_uri, dest_path))
-kfserving.Storage.download(src_uri, dest_path)
\ No newline at end of file
+
+with tempfile.TemporaryDirectory(prefix=".", dir=dest_path) as tmp_dir:
+    logging.info(f"Downloading staging directory: {tmp_dir}")
+    kfserving.Storage.download(src_uri, tmp_dir)
+    files = os.listdir(tmp_dir)
+    logging.info(f"Files downloaded: {files}")
+    logging.info(f"Copying to target location: {dest_path}")
```

Comment:
Does the main container currently have retry logic when dest_path is still empty and main container already starts running?

Reply:
I think @akirillov said it will just crashloop until the data is there. But while not super elegant, I think it's fine.

Reply:
From what I've observed, the main container will crashloop, however, the failure semantics could differ for different frameworks. E.g. KFServing uses vanilla TFServing server but provides wrappers for all other frameworks. Ideally, model downloading should be done in the server code without involving storage initializer at all. TFServing [supports](https://github.com/tensorflow/examples/blob/master/community/en/docs/deploy/s3.md) that, but all the wrappers deal with local copy as e.g. here: https://github.com/mesosphere/kfserving/blob/master/python/pytorchserver/pytorchserver/model.py#L36